sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -248,6 +253,38 @@ class Analyzer(
     }
   }
 
+  object ResolvePivot extends Rule[LogicalPlan] {
+    def apply(plan: LogicalPlan): LogicalPlan = plan transform {
+      case p: Pivot if !p.childrenResolved => p
+      case Pivot(groupByExprs, pivotColumn, pivotValues, aggregates, child) =>
+        val singleAgg = aggregates.size == 1
+        val pivotAggregates: Seq[NamedExpression] = pivotValues.flatMap{ value =>
+          aggregates.map{ aggregate =>
+            val filteredAggregate = aggregate.transformDown{
+              case u: UnaryExpression if u.isInstanceOf[AggregateExpression] =>
+                u.withNewChildren(Seq(
+                  If(EqualTo(pivotColumn, Literal(value)), u.child, Literal(null))
```

Comment:
I guess the underlying assumption of this line is that aggregate function should ignore `null` input values, right? (Basically, `null` should have no impact on the result of this aggregate function.) If so, can we add a comment at here?